modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotMessageCounter.java

```diff
@@ -56,6 +56,16 @@
 
     private static final int SLOT_SUBMIT_LOOP_SKIP_COUNT_THRESHOLD = 10;
 
+    /**
+     * Time between successive slot submit scheduled tasks.
+     */
+    public static final int SLOT_SUBMIT_TIMEOUT = 4000;
```

Comment:
In a slow message publishing scenario, this is a delay for each message for delivery. If we publish one message per minute then each message will have to wait till this timeout before it goes through the delivery part. A guaranteed delay for a message in slow message publishing scenario.